assets/src/components/v2/departures/later_departures.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import _ from "lodash";
+import cx from "classnames";
+import weakKey from "weak-key";
+
+import { hasOverflowX } from "Util/util";
+
+import DepartureRow from "./departure_row";
+import { Row } from "./normal_section";
+import RoutePill from "./route_pill";
+
+const MAX_LATER_DEPARTURES = 5;
+
+const LaterDepatures = ({ rows }: { rows: Row[] }) => {
+  const [limit, setLimit] = useState(MAX_LATER_DEPARTURES);
+  const departures = useMemo(
+    () =>
+      _.take(
+        rows.filter(
+          (r): r is DepartureRow & { type: "departure_row" } =>
+            r.type === "departure_row",
+        ),
+        limit,
+      ),
+    [rows, limit],
+  );
+
+  const ref = useRef<HTMLDivElement>(null);
+  const [idx, setIdx] = useState(0);
+
+  useLayoutEffect(() => {
+    if (hasOverflowX(ref)) {
+      setLimit(Math.max(0, limit - 1));
+    }
+  }, [limit]);
+
+  useEffect(() => {
+    if (departures.length < 2) return;
+    const refreshMs = Math.min(3750, 15000 / departures.length);
+
+    const interval = setInterval(() => {
+      setIdx((i) => (i + 1) % departures.length);
+    }, refreshMs);
+
+    return () => clearInterval(interval);
+  }, [departures.length]);
+
+  const departure = departures[idx % departures.length];
+
+  return (
+    <div className="later-departures">
+      <div className="later-departures__header" ref={ref}>
+        <h3>Later Departures</h3>
+        <ol>
+          {departures.map((d) => (
+            <li
+              key={weakKey(d)}
+              className={cx({
+                "later-departures--departure--selected": d === departure,
```

Comment:
BEM-wise this would probably be like... `later-departures__route--selected`? Not gonna stress too much about the exact naming, though, feel free to ignore 🙂

Reply:
Happy to make that change. I've never really used a class name convention like this. I've been a CSS modules (or similar) advocate for a long time but I can learn new tricks!

Reply:
Hah, I think CSS modules would more qualify as the "new thing", at least by my reckoning of how old BEM is at this point. I think I've already said so in another PR but I would also be interested in how we might adopt CSS modules, if and when the opportunity arises.

My understanding: BEM is a naming convention (family of conventions?) that divides UI into Blocks, which contain Elements, which may have Modifiers. Blocks are complete reusable components, Elements are the individual parts of them, Modifiers are different "states" or variants the same block/element could have. Classes are then named like `block-name__element-name--modifier-name` (if there are multiple modifiers, each gets its own class, with the full block/element name repeated for each one). There's other nuances, but it seems to me this project uses BEM less as part of an opinionated system and more just as a handy way to avoid global namespace collisions.